eman2/protocols/protocol_tomo_template_match.py

```diff
@@ -0,0 +1,198 @@
+
+        form.addParallelSection(threads=1, mpi=1)
+
+    # --------------------------- INSERT steps functions ----------------------
+
+    def _insertAllSteps(self):
+        self._insertFunctionStep('tempMatchStep')
+        self._insertFunctionStep("createOutputStep")
+
+
+    # --------------------------- STEPS functions -----------------------------
+
+    def tempMatchStep(self):
+
+        if self.boxSize.get()==-1:
+            self.box = self.ref.get().getDim()[0]
+        else:
+            self.box = self.boxSize.get()
+
+        volFile = os.path.abspath(self.ref.get().getFileName())
+        params = ""
+
+        for tomo in self.inputSet.get():
+            params = params + " %s" % tomo.getFileName()
+
+        params = params + " --reference=%s --nptcl=%d --dthr=%f --vthr=%f --delta=%f --sym=%s " \
+                          "--rmedge --rmgold --boxsz=%d" % (volFile, self.nptcl.get(), self.dthr.get(),
+                          self.vthr.get(), self.delta.get(), self.sym.get(), self.box)
+
+        program = eman2.Plugin.getProgram("e2spt_tempmatch.py")
+
+        self.runJob(program, params, cwd=os.path.abspath(self._getTmpPath()),
+                    env=eman2.Plugin.getEnviron())
+
+        #Move output files to Extra Path
+        moveFile(self._getTmpPath("ccc.hdf"),self._getExtraPath("particles" + ".hdf"))
+
+        for tomo in self.inputSet.get():
+            tomoName = os.path.basename(tomo.getFileName())
+            tomoName = os.path.splitext(tomoName)[0]
+            tomoCoord = "extra-" + tomoName + "_info.json"
+            moveFile(self._getTmpPath(os.path.join("info", tomoCoord)),
+                     self._getExtraPath("extra-" + tomoName + "_info.json"))
+
+        cleanPath(self._getTmpPath())
+
+    # --------------------------- INFO functions ------------------------------
+    def _validate(self):
+        errors = []
+        return errors
+
+    # --------------------------- UTILS functions ------------------------------
+    def getSummary(self, coord3DSet):
+        summary = []
+        summary.append("Number of particles picked: %s" % coord3DSet.getSize())
+        summary.append("Particle size: %s" % coord3DSet.getBoxSize())
+        return "\n".join(summary)
+
+    def _readSetOfCoordinates3D(self, jsonBoxDict, coord3DSetDict, inputTomo):
+        if jsonBoxDict.has_key("boxes_3d"):
+            boxes = jsonBoxDict["boxes_3d"]
+
+            for box in boxes:
+                classKey = box[5]
+                coord3DSet = coord3DSetDict[classKey]
+                coord3DSet.enableAppend()
+
+                self._readCoordinates3D(box, coord3DSet, inputTomo)
+
+    def _readCoordinates3D(self, box, coord3DSet, inputTomo):
+        x, y, z = box[:3]
+        coord = Coordinate3D()
+        coord.setPosition(x, y, z)
+        coord.setVolume(inputTomo)
+        coord3DSet.append(coord)
+
+
+    def createOutputStep(self):
+
+        # Create a Set of 3D Coordinates per class
```

Comment:
With this comment I understand that you can pick different classes of particles at the same time. If this is true, you should use a different template for each class. But I think now is not possible to select more than one template as input. Am I right or I am missunderstanding something?